modules/renter/proto/negotiate.go

```diff
@@ -66,38 +66,54 @@ func verifySettings(conn net.Conn, host modules.HostDBEntry) (modules.HostDBEntr
 	return host, nil
 }
 
-// verifyRecentRevision confirms that the host and contractor agree upon the current
-// state of the contract being revised.
-func verifyRecentRevision(conn net.Conn, contract contractHeader, hostVersion string) error {
+// getRecentRevision downloads the current revision from the host
+// and checks signatures.
+func getRecentRevision(conn net.Conn, fcid types.FileContractID, sk crypto.SecretKey, windowStart types.BlockHeight, hostVersion string) (types.FileContractRevision, error) {
```

Comment:
I don't understand why it's necessary to change `verifyRecentRevision`. It doesn't look like you're using `lastRevision` in `GetMetadata`, so why do you need to return it?

Reply:
The caller of `GetMetadata` can calculate merkle root of sector IDs returned by `GetMetadata` and compare it with merkle root from lastRevision. This is a way to verify that sector IDs are correct for some revision (host still can return old revision).